src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests4.cs

```diff
@@ -1489,6 +1731,557 @@ static int M1(bool? b1, bool? b2)
                 );
         }
 
+        [Fact]
+        public void DeconstructVsITuple_01()
+        {
+            // From LDM 2018-11-05:
+            // 1. If the type is a tuple type (any arity >= 0; see below), then use the tuple semantics
+            // 2. If "binding" a Deconstruct invocation would find one or more applicable methods, use Deconstruct.
+            // 3. If the type satisfies the ITuple deconstruct constraints, use ITuple semantics
+            // Here we test the relative priority of steps 2 and 3.
+            // - Found one applicable Deconstruct method (even though the type implements ITuple): use it
+            var source = @"using System;
+using System.Runtime.CompilerServices;
+class Program
+{
+    static void Main()
+    {
+        IA a = new A();
+        if (a is (var x, var y))  // tuple pattern containing var patterns
+            Console.Write($""{x} {y}"");
+    }
+}
+interface IA : ITuple
+{
+    void Deconstruct(out int X, out int Y);
+}
+class A: IA, ITuple
+{
+    void IA.Deconstruct(out int X, out int Y) => (X, Y) = (3, 4);
+    int ITuple.Length => 2;
```

Comment:
>2; [](start = 25, length = 2)

Consider throwing for members that don't participate in execution. (also applies in other tests below)